components/web_discovery/browser/rsa.cc

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/rsa.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/threading/thread_restrictions.h"
+#include "crypto/rsa_private_key.h"
+#include "crypto/signature_creator.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr size_t kRsaKeySize = 2048;
+
+}  // namespace
+
+GeneratedRSAKey::GeneratedRSAKey() = default;
+GeneratedRSAKey::~GeneratedRSAKey() = default;
+
+EncodedRSAKeyPair::EncodedRSAKeyPair() = default;
+EncodedRSAKeyPair::~EncodedRSAKeyPair() = default;
+
+ImportedRSAKey::ImportedRSAKey() = default;
+ImportedRSAKey::~ImportedRSAKey() = default;
+
+std::unique_ptr<GeneratedRSAKey> GenerateRSAKey() {
```

Comment:
it seems like these should be static `Create` methods and make the constructors private/friend so they are only ever created using these methods? Otherwise it seems like it defeats the purpose of separating this code since someone could circumvent it by manually creating them?